apps/backend/src/routes/fileupload/+page.svelte

```diff
@@ -0,0 +1,170 @@
+<script lang="ts">
+	import { firestore } from '$lib/firebase';
+	import { collection, addDoc, getFirestore } from 'firebase/firestore';
+	import { storage } from '$lib/firebase';
+	import type { FirebaseStorage } from 'firebase/storage';
+	import { getStorage, ref, uploadBytes } from 'firebase/storage';
+	import { Form, Field, Select, ErrorMessage, createForm } from 'svelte-forms-lib';
+	import * as yup from 'yup';
+	import type { FirebaseApp } from 'firebase/app';
+	import { v4 } from 'uuid';
+
+	interface FormValues {
+		fileUpload: FileList | undefined;
+		fileClass: string;
+		isChecked: boolean;
+		fileEncryptionHash: string;
+		fileExtension: string;
+		fileId: string;
+		fileName: string;
+		fileOwnerId: string;
+	}
+
+	const { form, errors, handleChange, handleSubmit } = createForm<FormValues>({
+		initialValues: {
+			fileUpload: undefined,
+			fileClass: 'Select file class',
+			isChecked: false,
+			fileEncryptionHash: '',
+			fileExtension: '',
+			fileId: '',
+			fileName: '',
+			fileOwnerId: ''
+		},
+		validationSchema: yup.object({
+			fileUpload: yup.string().required('File is required!'),
+			fileClass: yup.string().oneOf(['TOPSECRET', 'SENSITIVE', 'OPEN']),
+			isChecked: yup
+				.boolean()
+				.oneOf(
+					[true],
+					'Please ensure you have selected the right classification by checking the checkbox.'
+				)
+		}),
+		onSubmit: async (data) => {
+			console.log(data);
+			var uniqueId = v4();
+
+			if (!data.fileUpload || data.fileClass === 'Select file class' || !data.isChecked) {
+				console.error('Missing required data!');
+				return; // Early return if fields are missing
+			}
+
+			const file = data.fileUpload[0];
+			const fileArrayBuffer = await file.arrayBuffer();
+			uploadBytes(ref(getStorage(), `armadillo-files/${uniqueId}`), fileArrayBuffer)
+				.then(console.log)
+				.catch(console.error);
+			// const fileData = Buffer.from(await data.fileUpload![0].arrayBuffer());
+			// console.log(fileData.toString('base64'));
+
+			const docRef = await addDoc(collection(firestore, 'files'), {
```

Comment:
Use `setDoc` instead to explicitly set the document id in Firestore, since we are using uuidv4. Currently, we are relying on Firestore to automatically create a file id for us, which is not our intended use case